test/parallel/test-http-outgoing-implicit-header.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+require('../common');
+
+const assert = require('assert');
+const http = require('http');
+
+const server = http.createServer((req, res) => {
+  res._hasBody = false;
```

Comment:
Never use private methods when creating tests.

Reply:
I don't think you need a new test here, just modify test/parallel/test-http-head-response-has-no-body-end.js to test this new behavior.

Reply:
thank you very much for your suggestion, I will implement it